File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.c
int longestSubarray(int* nums, int numsSize) {
    bool flag = false;
    int one = 0;
    int two = 0;
    int answer = 0;

    for (int i = 0; i < numsSize; i++) {
        if (nums[i] == 0) {
            if (one + two > answer) {
                answer = one + two;
            }
            one = two;
            two = 0;
            flag = true;
        } else {
            two++;
        }
    }
    if (one + two > answer) {
        answer = one + two;
    }
    
    if (flag == false) {
        answer--;
    }
    
    return answer;
}

File: 1586-longest-subarray-of-1s-after-deleting-one-element/1586-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        flag = False
        one, two = 0, 0
        answer = 0
        for i, num in enumerate(nums):
            if num == 0:
                answer = max(answer, one + two)
                one, two = two, 0
                flag = True
            else:
                two += 1
        
        answer = max(answer, one + two)

        if not flag:
            answer -= 1

        return answer
                

